testbed/testbed/mock_backend.go

```diff
@@ -149,6 +159,67 @@ func (mb *MockBackend) ConsumeMetric(md pdata.Metrics) {
 	if mb.isRecording {
 		mb.ReceivedMetrics = append(mb.ReceivedMetrics, md)
 	}
+	if mb.isScraping {
+		currentTimestamp := getMetricTimestamp(md)
+		receivedTimestampsCount := len(mb.ReceivedTimestamps)
+		if receivedTimestampsCount == 0 || !mb.ReceivedTimestamps[receivedTimestampsCount-1].Equal(currentTimestamp) {
+			mb.ReceivedTimestamps = append(mb.ReceivedTimestamps, currentTimestamp)
+		}
+	}
+}
+
+func getMetricTimestamp(md pdata.Metrics) time.Time {
```

Comment:
Does this return the timestamp of the very first metric data point? If yes then getFirstMetricTimestamp may be a better name and some doc comment to explain what the func does would also be useful.